obs/packages.go

```diff
@@ -0,0 +1,80 @@
+package obs
+
+import (
+	"context"
+	"encoding/xml"
+	"fmt"
+	"net/http"
+	"net/url"
+)
+
+// ListPackages lists the packages of a given OBS project
+func (o *OBS) ListPackages(ctx context.Context, projectName string) (*Package, error) {
+	urlPath, err := url.JoinPath(o.options.APIURL, "source", projectName)
+	if err != nil {
+		return nil, fmt.Errorf("listing obs packages: joining url: %w", err)
+	}
+
+	resp, err := o.InvokeOBSEndpoint(ctx, http.MethodGet, urlPath, nil)
+	if err != nil {
+		return nil, fmt.Errorf("listing obs packages: invoking obs endpoint: %w", err)
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		var status Status
+		if err := xml.NewDecoder(resp.Body).Decode(&status); err != nil {
+			return nil, &APIError{
+				HTTPStatusCode: resp.StatusCode,
+				OBSStatusCode:  "",
+				Message:        fmt.Sprintf("listing obs packages: decoding error response: %v", err),
+			}
+		}
+
+		return nil, &APIError{
+			HTTPStatusCode: resp.StatusCode,
+			OBSStatusCode:  status.Code,
+			Message:        status.Summary,
+		}
+	}
+
+	var pkg Package
```

Comment:
This could be a pointer
```suggestion
	pkg := &Package{}
```